tower/src/retry/mod.rs

```diff
@@ -17,9 +17,20 @@ pin_project! {
     /// Configure retrying requests of "failed" responses.
     ///
     /// A [`Policy`] classifies what is a "failed" response.
+    ///
+    /// # Clone
+    ///
+    /// This middleware requires that the `Service` is `Clone` due to requirements
+    /// of `'static` futures. To easily add `Clone` to your service you can
+    /// use the `Buffer` middleware.
+    ///
+    /// The `Policy` is also required to implement `Clone`. This middleware will
+    /// clone the policy for each _request session_. This means one instance
+    /// of the policy will exist for one request and its subsequent retries only.
+    /// This means the mutable reference is only for that session and if you want
+    /// to share data between request sessions then you will need to `Arc<Mutex<...>`.
```

Comment:
how about:
```suggestion
    /// clone the policy for each _request session_. This means a new clone
    /// of the policy will be created for each initial request and any subsequent 
    /// retries of that request. Therefore, any state stored in the `Policy` instance
    /// is for that request session only. In order to share data across request 
    /// sessions, that shared state may be stored in an [`Arc`], so that all clones
    /// of the `Policy` type reference the same instance of the shared state.
    ///
    /// [`Arc`]: std::sync::Arc
```